### DISCUSSION TITLE
Inofficial Zigbee Router firmware released for ITead Sonoff ZBBridge as alternative when used with Tasmota and/or ESPHome

### DISCUSSION THREAD

**Discussion by Hedda** [Koenkk/zigbee2mqtt#11461] [General]
FYI, xsp1989 has uploaded a signed Zigbee Router firmware for ITead Sonoff ZBBridge, that could make it possible to make it into a network-attached Tasmota/ESPHome controlled Zigbee Router device (instead of as a remote Zigbee Coordinator coordinator)?

> [https://drive.google.com/file/d/1H-M5CTh_XGGVl2te4vHLQ5SfGPHL6yCS/view?usp=sharing](https://drive.google.com/file/d/1H-M5CTh_XGGVl2te4vHLQ5SfGPHL6yCS/view?usp=sharing)
> 
> This is the signed routing firmware used by ZBB, the usage method is the same as the unsigned firmware.
> 
> [https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/Zigbee3.0_Dongle/RouterForDongle/README.md](https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/Zigbee3.0_Dongle/RouterForDongle/README.md)

Do you think Tasmota/ESPHome can send basic serial commands to this Zigbee Router firmware to initiate paring/joining and restart/reset?

The idea around this concept is also discussed for ITead's Sonoff ZBBridge hacked with Tasmota/ESPHome here:

[https://github.com/arendst/Tasmota/discussions/11536](https://github.com/arendst/Tasmota/discussions/11536)

[https://github.com/thegroove/esphome-zbbridge/issues/3](https://github.com/thegroove/esphome-zbbridge/issues/3)

[https://github.com/syssi/esphome-zb-gw03/issues/12](https://github.com/syssi/esphome-zb-gw03/issues/12)

[https://github.com/xsp1989/zigbeeFirmware/issues/2](https://github.com/xsp1989/zigbeeFirmware/issues/2)

[https://github.com/xsp1989/zigbeeFirmware/issues/16](https://github.com/xsp1989/zigbeeFirmware/issues/16)

I have not tried this myself as I got rid of my ITead Sonoff ZBBridge because being WiFi-based it did not work any good as a remote Zigbee Coordinator adapter, but might reconsider buying one (or a ZB-GW03 with Ethernet port) if it can work as a Zigbee Router device that can be paired/joined and restarted/reset remotely via Tasmota and/or ESPHome. So that Tasmota/ESPHome is basically only used to initiate virtual join/pair button and restart/reset the device similar to a remote power-cycle it hangs.

**Comment by Hedda** [comment]
Update from xsp1989 on router firmware posted on https://github.com/arendst/Tasmota/discussions/11536

"This firmware can ground the PA00 5S into pairing mode, and ground the RST to restart routing. UART is used to transmit data transparently.

and

"There is also another version of the routing firmware that can control the join network through a proprietary serial port protocol"

**Comment by Hedda** [comment]
Check out this new "How to Convert the Sonoff Zigbee Bridge into a Router / Repeater" article by @digiblur

https://www.digiblur.com/2022/02/convert-sonoff-zigbee-bridge-to-router-repeater.html

**Reply by ZamliyMedre** [reply]
I have installed this firmware on 2 of the Sonoff bridges connecting to a Sonoff Bridge coordinator again flashed with tasmota.... unfortunately, as it is router firmware seems to only let maximum of 6 child device connection per router.

Not sure why, as using the tasmota bridge firmware it can support upto 32 devices? same hardware why its limited using router I do not understand.

I have opened an issue regards to this in github .. hopefully developers maybe able to provide some direction...
